content/explugins/on.md

```diff
@@ -0,0 +1,66 @@
++++
+title = "on"
+description = "*on* executes a command when a specified event is triggered."
+weight = 10
+tags = [  "plugin" , "on" ]
+categories = [ "plugin", "external" ]
+date = "2018-01-22T07:53:19+01:00"
+repo = "https://github.com/mholt/caddy"
+home = "https://caddyserver.com/docs/onevent"
+enabled = "default"
++++
+
+## Description
+
+*on* executes a command when a specified event is triggered. This can be useful for preparing to
+ serve a site by running a script or starting a background process when the server starts, or for
+ stopping it when the server exits.
+
+Each command that is executed is blocking, unless you suffix the command with a space and &, which
+will cause the command to be run in the background. (Do not do this when the server is exiting, or
+the command may not finish before its parent process exits.) The output and error of the command go
+to standard outpu and standard error, respectively. There is no standard input.
```

Comment:
outpu -> output

Reply:
Actually, standard input is also mapped to stdin (https://github.com/mholt/caddy/blob/master/onevent/hook/hook.go#L27). I tried running coredns with "on startup sudo touch /tmp/file" and I was asked for password, I entered it and the file was successfully created with root access rights. So, seems stdin works fine. Another question is how convenient it may be for running a server application